lib/api.js

```diff
@@ -42,6 +42,62 @@ module.exports = {
         })
     },
 
+    getLastTransaction: function (address, endpoint) {
+        return fetch(endpoint + "/api", {
+            method: "POST",
+            headers: {
+                'Content-Type': 'application/json',
+                'Accept': 'application/json',
+            },
+            body: JSON.stringify({
+                query: `query {
+                    lastTransaction(address: "${address}") {
+                        address
+                    }
+                }`
+            })
+        })
+        .then(r => r.json())
+        .then((res) => {
+            if (res.data.lastTransaction == null) {
+                return ""
+            }
+            else {
+                return res.data.lastTransaction.address
+            }
+        })
+    },
+
+    getTransactionChain: function (address, endpoint) {
```

Comment:
Here is the same, you only fetch the addresses, but your function name signify to load the all transaction for the chain.